tests/e2e/state.go

```diff
@@ -519,28 +534,48 @@ func (tr TestConfig) getValPower(chain ChainID, validator ValidatorID) uint {
 	}
 
 	for _, val := range valset.Validators {
-		if val.Address == tr.validatorConfigs[validator].ValconsAddress ||
-			val.Address == tr.validatorConfigs[validator].ConsumerValconsAddress {
-
-			votingPower, err := strconv.Atoi(val.VotingPower)
-			if err != nil {
-				log.Fatalf("strconv.Atoi returned an error while converting validator voting power: %v, voting power string: %s, validator set: %s", err, val.VotingPower, pretty.Sprint(valset))
+		if chain == ChainID("provi") {
+			// use binary with Bech32Prefix set to "cosmos"
+			if val.Address != tr.validatorConfigs[validator].ValconsAddress {
+				continue
 			}
+		} else {
+			// use binary with Bech32Prefix set to "consumer"
+			if val.Address != tr.validatorConfigs[validator].ValconsAddressOnConsumer &&
+				val.Address != tr.validatorConfigs[validator].ConsumerValconsAddress {
+				continue
+			}
+		}
 
-			return uint(votingPower)
+		votingPower, err := strconv.Atoi(val.VotingPower)
+		if err != nil {
+			log.Fatalf("strconv.Atoi returned an error while converting validator voting power: %v, voting power string: %s, validator set: %s", err, val.VotingPower, pretty.Sprint(valset))
 		}
+
+		return uint(votingPower)
 	}
 
 	// Validator not in set, its validator power is zero.
 	return 0
 }
 
 func (tr TestConfig) getValStakedTokens(chain ChainID, validator ValidatorID) uint {
+	valoperAddress := tr.validatorConfigs[validator].ValoperAddress
+	if chain != ChainID("provi") {
```

Comment:
I wonder if it makes sense for this repetitive logic to be in a util function. I know it's often slightly different, and it seems annoying to extract a core logic (maybe it's more like 4 different core logics, but then it might not make sense to extract at all). Just wondering out loud, not sure I have a strong feeling either way

Reply:
@p-offtermatt I could add a struct that looks like this
```golang
type Bech32Address struct {
	// Provider address marshaled to string using Bech32 with ProviderAccountPrefix
	ProviderAddress string
	// Provider address marshaled to string using Bech32 with ConsumerAccountPrefix
	ProviderAddressOnConsumer string
	// Consumer address marshaled to string using Bech32 with ConsumerAccountPrefix
	ConsumerAddress string
	// Consumer address marshaled to string using Bech32 with ProviderAccountPrefix
	ConsumerAddressOnProvider string
}
```
Then have in the ValidatorConfig struct the following fields
```golang
	// Validator account address
	AccountAddress Bech32Address
	// Validator operator address
	ValoperAddress Bech32Address
	// Validator consensus address
	ValconsAddress Bech32Address
```
And move the logic to a method like this
```golang
func (bech32Addr Bech32Address) ChooseAddress(chain ChainID, useConsumerKey bool) string {
	addr := bech32Addr.ProviderAddress
	if chain != ChainID("provi") {
		// use binary with Bech32Prefix set to ConsumerAccountPrefix
		if useConsumerKey {
			addr = bech32Addr.ConsumerAddress
		} else {
			addr = bech32Addr.ProviderAddressOnConsumer
		}
	}
	return addr
}
```

However, this would refactor a lot of the code and make the reviewing harder. In case you think it's a good idea, I would prefer to do that in a separate PR.

Reply:
It is used in a different context. Look for `ConsumerValconsAddressOnProvider` in this PR.